cmd/minikube/cmd/disk-usage.go

```diff
@@ -0,0 +1,128 @@
+/*
+Copyright 2016 The Kubernetes Authors All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/spf13/cobra"
+
+	"k8s.io/klog/v2"
+	"k8s.io/minikube/pkg/minikube/out"
+)
+
+var (
+	longOutFlag  bool
+	rootOnlyFlag bool
+)
+
+var decimalUnitPrefixMap = map[int]string{
+	0: "",
+	1: "K",
+	2: "M",
+	3: "G",
+	4: "T",
+	5: "P",
+	6: "E",
+	7: "Z",
+	8: "Y",
+}
+
+// runDiskUsage represents the disk-usage command
+var diskUsageCmd = &cobra.Command{
+	Use:   "disk-usage",
+	Short: "Shows current disk usage of minikube",
+	Long:  "Shows current disk usage of minikube",
+	Run:   runDiskUsage,
+}
+
+func runDiskUsage(cmd *cobra.Command, args []string) {
+	minikubeDirs := []string{dirs[0]}
+	if !rootOnlyFlag {
+		minikubeSubDirs, err := os.ReadDir(dirs[0])
+		if err != nil {
+			klog.Errorf("Error reading minikube directories:\n%s", err)
+		}
+		for _, file := range minikubeSubDirs {
+			if file.IsDir() {
+				minikubeDirs = append(minikubeDirs, fmt.Sprintf("%s/%s", dirs[0], file.Name()))
+			}
+		}
+	}
+	for _, dir := range minikubeDirs {
+		dirNameList := strings.Split(dir, "/")
+		dirName := dirNameList[len(dirNameList)-1]
+		info, err := os.Lstat(dir)
+		if err != nil {
+			klog.Errorf("Error reading info about configured minikube path:\n%s", err)
+			os.Exit(1)
+		}
+		totalFileSize := diskUsage(dir, info)
+		totalFileSizeOut := strconv.FormatInt(diskUsage(dir, info), 10)
+
+		var decimalUnitPrefixMapKey int
+		if !longOutFlag {
+			totalFileSizeOut, decimalUnitPrefixMapKey = int64ToFloat32FormattedString(totalFileSize)
+		}
+
+		out.Infof("{{.directory}} - {{.diskSize}}{{.decimalUnitPrefix}}B", out.V{"directory": dirName, "diskSize": totalFileSizeOut, "decimalUnitPrefix": decimalUnitPrefixMap[decimalUnitPrefixMapKey]})
+	}
+}
+
+func int64ToFloat32FormattedString(in_int int64) (string, int) {
```

Comment:
See also github.com/docker/go-units

Reply:
![image](https://user-images.githubusercontent.com/13648427/145096681-64b58232-f7a7-4818-9d1b-bdfb5db4ad32.jpeg)

Reply:
The biggest fight is between the mebibytes and the megabytes